src/real_double.cpp

```diff
@@ -0,0 +1,42 @@
+/**
+ *  \file RealDouble.h
+ *  Class for RealDouble built on top of Number class
+ *
+ **/
+#include "basic.h"
+#include "real_double.h"
+
+namespace SymEngine {
+
+RealDouble::RealDouble(double i) {
+    this->i = i;
+}
+
+std::size_t RealDouble::__hash__() const
+{
+    std::hash<double> hash_fn;
+    // only the least significant bits that fit into "signed long int" are
+    // hashed:
+    return hash_fn(i);
+}
+
+bool RealDouble::__eq__(const Basic &o) const
+{
+    if (is_a<RealDouble>(o)) {
+        const RealDouble &s = static_cast<const RealDouble &>(o);
+        return this->i == s.i;
+    }
+    return false;
```

Comment:
I think it's fine as it is --- the only defined comparison is if the double is the same variable, perhaps pointed to from two different pointers. Which is what you are doing. I think that's fine, otherwise it would be confusing if you did:
```
a = 1.0
b = a
```
and then `a == b` would return False.